app/src/main/java/com/example/vshcheglov/webshop/presentation/main/MainViewModel.kt

```diff
@@ -0,0 +1,180 @@
+package com.example.vshcheglov.webshop.presentation.main
+
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.example.vshcheglov.webshop.App
+import com.example.vshcheglov.webshop.data.DataProvider
+import com.example.vshcheglov.webshop.domain.Basket
+import com.example.vshcheglov.webshop.domain.Product
+import com.example.vshcheglov.webshop.presentation.helpres.Event
+import com.example.vshcheglov.webshop.presentation.helpres.EventWithContent
+import com.example.vshcheglov.webshop.presentation.main.helpers.SearchFilter
+import kotlinx.coroutines.*
+import timber.log.Timber
+import java.util.*
+import javax.inject.Inject
+
+class MainViewModel : ViewModel() {
+    @Inject
+    lateinit var dataProvider: DataProvider
+
+    private var isNeedToSaveAvatar = false
+
+    private val job: Job = Job()
+    private val uiCoroutineScope: CoroutineScope = CoroutineScope(Dispatchers.Main + job)
+
+    private val _stateLiveData: MutableLiveData<MainViewState> = MutableLiveData<MainViewState>().apply {
+        value = MainViewState()
+    }
+    val stateLiveData: LiveData<MainViewState> = _stateLiveData
+
+    private val _commandLiveData: MutableLiveData<EventWithContent<MainCommand>> = MutableLiveData()
+    val commandLiveData: LiveData<EventWithContent<MainCommand>> = _commandLiveData
+
+    private fun getState() = stateLiveData.value!!
+
+    init {
+        App.appComponent.inject(this)
+
+        try {
```

Comment:
Not a good approach to call everything in constructor or init block. Revise please